src/test_report.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Jeny Sadadia
+# Author: Jeny Sadadia <jeny.sadadia@gmail.com>
+
+import os
+import sys
+
+import kernelci.config
+import kernelci.data
+from kernelci.cli import Args, Command, parse_opts
+import jinja2
+
+
+class TestReport:
+
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.data.get_db(self._db_config, api_token)
+
+    def run(self):
+        sub_id = self._db.subscribe('node')
+        self._print("Listening for test completion events")
+        self._print("Press Ctrl-C to stop.")
+
+        try:
+            while True:
+                sys.stdout.flush()
+                event = self._db.get_event(sub_id)
+
+                node = self._db.get_node_from_event(event)
+                if node['status'] is None:
+                    continue
+
+                self._print(f"Node: {node}")
+
+                root_node = node
+                if root_node['revision']['describe'] is None:
+                    root_node['revision']['describe'] = "(Unknown)"
```

Comment:
I think this can be done with a default value in the template instead, as the text to show when there's no value is more something that should be part of the rendering than the logic to supply the data used in the report.

Reply:
Also it's not great to replace the node data as other parts of the code might be relying on the real value which is `None`.

Reply:
This is fixed.